src/Database/Query.php

```diff
@@ -229,6 +236,40 @@ public function getConnection(): Connection
         return $this->_connection;
     }
 
+    /**
+     * Overrides the default connection role.
+     *
+     * @param string $role Connection role
+     * @return $this
+     */
+    public function setConnectionRole(string $role)
+    {
+        $this->connectionRole = $role;
+
+        return $this;
+    }
+
+    /**
+     * Returns the connection role for the query.
+     *
+     * Defaults to Connection::ROLE_REPLICA for select queries.
+     * You can override the default using setConnectionRole().
+     *
+     * @return string
+     */
+    public function getConnectionRole(): string
+    {
+        if ($this->connectionRole !== null) {
```

Comment:
The property can default to `primary` and avoid this check.

Reply:
Then it would always be primary for select statements, right?

Reply:
We could always default to `primary` and allow users to opt into running queries on the replica as required.  It is basically impossible for us to determine which reads are safe to run on the replica and which are not.

Reply:
Ok, we'll let users decided. We can possibly set default roles in Table.